cms/djangoapps/contentstore/tasks.py

```diff
@@ -391,19 +396,72 @@ def import_olx(self, user_id, course_key_string, archive_path, archive_name, lan
     """
     Import a course or library from a provided OLX .tar.gz archive.
     """
-    set_code_owner_attribute_from_module(__name__)
+    current_step = 'Unpacking'
     courselike_key = CourseKey.from_string(course_key_string)
-    try:
-        user = User.objects.get(pk=user_id)
-    except User.DoesNotExist:
-        with translation_language(language):
-            LOGGER.error(f'Course import {courselike_key}: Unknown User ID: {user_id}')
-            self.status.fail(_('Unknown User ID: {0}').format(user_id))
+    set_code_owner_attribute_from_module(__name__)
+    set_custom_attributes_for_course_key(courselike_key)
+    log_prefix = f'Course import {courselike_key}'
+    self.status.set_state(current_step)
+
+    # Locate the uploaded OLX archive (and download it from S3 if necessary)
+    # Do everything in a try-except block to make sure everything is properly cleaned up.
```

Comment:
Since this has been refactored, you might want to take a look at this comment and move it to appropriate place.